src/Compilers/Test/Core/Mocks/Silverlight.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Roslyn.Test.Utilities;
+
+/// <summary>
+/// The assemblies produced here are designed to mimic the public key token structure of 
+/// silverlight references. This often presents challenges to the compiler because it has
+/// to know that two mscorlib with different public key tokens need to be treated as the 
+/// identicial. The assemblies produced here have the same identity of those that come
+/// from silverlight but without necessarily the same type contents.
+/// </summary>
```

Comment:
I went back and forth on this decision several times. The Silverlight references are _only_ used right now to test assembly unification policies (that I could see). Rather than the overhead of bringing in more .dll files decided to just dynamically build the same assemblies that would let us test unification policy.